src/server/dflycmd.cc

```diff
@@ -359,14 +366,19 @@ void DflyCmd::TakeOver(CmdArgList args, ConnectionContext* cntx) {
   if (!sync_id)
     return;
 
-  unique_lock lk(replica_ptr->mu);
-  if (!CheckReplicaStateOrReply(*replica_ptr, SyncState::STABLE_SYNC, rb))
-    return;
+  {
+    shared_lock lk = replica_ptr->GetSharedLock();
+    if (!CheckReplicaStateOrReply(*replica_ptr, SyncState::STABLE_SYNC, rb))
+      return;
```

Comment:
A small idea would be moving the ToReply part out so it doesn't happen under a lock

Reply:
I think, it's actually a good idea because we may theoretically stuck on send if say the socket is full. 
Since it's an admin socket, it's unlikely, so I will add a TODO